app/src/ui/history/compare.tsx

```diff
@@ -0,0 +1,477 @@
+    const compareState = this.props.compareState
+    const selectedCommit = this.state.selectedCommit
+    const commitSHAs = compareState.commitSHAs
+
+    const emptyListMessage =
+      compareState.formState.kind === ComparisonView.None
+        ? 'No history'
+        : 'No commits'
+
+    return (
+      <CommitList
+        gitHubRepository={this.props.repository.gitHubRepository}
+        commitLookup={this.props.commitLookup}
+        commitSHAs={commitSHAs}
+        selectedSHA={selectedCommit !== null ? selectedCommit.sha : null}
+        gitHubUsers={this.props.gitHubUsers}
+        localCommitSHAs={this.props.localCommitSHAs}
+        emoji={this.props.emoji}
+        onViewCommitOnGitHub={this.props.onViewCommitOnGitHub}
+        onRevertCommit={this.props.onRevertCommit}
+        onCommitSelected={this.onCommitSelected}
+        onScroll={this.onScroll}
+        emptyListMessage={emptyListMessage}
+      />
+    )
+  }
+
+  private renderActiveTab() {
+    const formState = this.props.compareState.formState
+    return (
+      <div className="the-commits">
+        {this.renderCommitList()}
+        {formState.kind === ComparisonView.Behind
+          ? this.renderMergeCallToAction(formState)
+          : null}
+      </div>
+    )
+  }
+
+  private renderFilterList() {
+    const compareState = this.props.compareState
+    return (
+      <BranchList
+        ref={this.onBranchesListRef}
+        defaultBranch={compareState.defaultBranch}
+        currentBranch={this.props.currentBranch}
+        allBranches={compareState.allBranches}
+        recentBranches={compareState.recentBranches}
+        filterText={this.state.filterText}
+        textbox={this.textbox!}
+        selectedBranch={this.state.focusedBranch}
+        canCreateNewBranch={false}
+        onSelectionChanged={this.onSelectionChanged}
+        onItemClick={this.onBranchItemClicked}
+        onFilterTextChanged={this.onBranchFilterTextChanged}
+        renderBranch={this.renderCompareBranchListItem}
+      />
+    )
+  }
+
+  private renderMergeCallToAction(formState: ICompareBranch) {
+    if (this.props.currentBranch == null) {
+      return null
+    }
+
+    const count = formState.aheadBehind.behind
+
+    return (
+      <div className="merge-cta">
+        <Button
+          type="submit"
+          disabled={count <= 0}
+          onClick={this.onMergeClicked}
+        >
+          Merge into <strong>{this.props.currentBranch.name}</strong>
+        </Button>
+
+        {this.a(formState)}
+      </div>
+    )
+  }
+
+  private a(formState: ICompareBranch) {
+    const branch = formState.comparisonBranch
+    const count = formState.aheadBehind.behind
+
+    if (count > 0) {
+      const pluralized = count === 1 ? 'commit' : 'commits'
+      return (
+        <div className="merge-message">
+          This will merge
+          <strong>{`${count} ${pluralized}`}</strong>
+          {` `}from{` `}
+          <strong>{branch.name}</strong>)
```

Comment:
I think this is the extraneous `)` that @meaghanlewis found. saw it on my computer, too. let's remove it